cpp/src/arrow/dataset/partition_test.cc

```diff
@@ -458,8 +458,10 @@ TEST_F(TestPartitioning, HivePartitioning) {
 
 TEST_F(TestPartitioning, HivePartitioningEquals) {
   const auto& array_vector = ArrayVector();
-  const auto& other_vector = {ArrayFromJSON(utf8(), R"(["foo", "bar", "baz"])"),
-                              ArrayFromJSON(utf8(), R"(["bar", "foo", "baz"])")};
+  std::vector<std::shared_ptr<arrow::Array>> other_vector;
+  other_vector.reserve(2);
```

Comment:
nit, but for consistency with the surrounding code, this could just be `ArrayVector other_vector(2)`